presto-common/src/test/java/com/facebook/presto/common/TestPage.java

```diff
@@ -40,14 +40,17 @@ public class TestPage
     @Test
     public void testGetRegion()
     {
-        assertEquals(new Page(10).getRegion(5, 5).getPositionCount(), 5);
+        Page page = new Page(10);
+        assertEquals(page.getRegion(5, 5).getPositionCount(), 5);
+        assertEquals(page.getRegion(0, 10).getPositionCount(), 10);
```

Comment:
Can you please add a test that shows that this is the same object?  I.e. `page.getRegion(0, 10)` === `page`

Reply:
have added Page page = new Page(10);
        Page region = page.getRegion(0, 10);
        assertEquals(page.getRegion(5, 5).getPositionCount(), 5);
        assertEquals(region.getPositionCount(), 10);
        assertSame(page, region);